modules/basegl/src/algorithm/volumenormalization.cpp

```diff
@@ -0,0 +1,71 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2016-2021 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/basegl/algorithm/volumenormalization.h>
+
+#include <modules/opengl/volume/volumegl.h>
+#include <modules/opengl/shader/shaderutils.h>
+#include <modules/opengl/volume/volumeutils.h>
+
+namespace inviwo {
+
+std::shared_ptr<Volume> VolumeNormalization::normalize(const Volume& volume, const int channel) {
+    auto outVolume = std::make_shared<Volume>(volume.getDimensions(), volume.getDataFormat(),
```

Comment:
Output should be float with only one channel.

Reply:
Not sure I agree. What I would rather do is to let the user specify which channels s/he wants to normalize. One could have a processor with a checkbox for every available channel. We could work with shader defines for control-free code.

I would argue against normalizing the data in sense of a norm since that would be more of a vectorfield operation.

I have added a corresponding processor in my module. Mostly because a volume processor that doesn't inherit from 
`VolumeGLProcessor` wouldn't fit in with the rest of the volume processors.

Reply:
There is no need for volume processors to inherit VolumeGlProcessor

Reply:
Output format must be float. Otherwise your data range below will not work